packages/rest/src/validation/request-body.validator.ts

```diff
@@ -102,22 +109,23 @@ function validateValueAgainstJsonSchema(
     allErrors: true,
   });
 
-  try {
-    if (ajv.validate(schemaWithRef, body)) {
-      debug('Request body passed AJV validation.');
-      return;
-    }
-  } catch (err) {
-    debug('Fails to execute AJV validation:', err);
-    // TODO: [rfeng] Do we want to introduce a flag to disable validation
-    // or sink validation errors?
-    throw err;
+  if (!validateSchema.has(jsonSchema)) {
+    const compiled = ajv.compile(schemaWithRef);
+    validateSchema.set(jsonSchema, compiled);
```

Comment:
Wouldn't `validateSchema` become unnecessarily large (which may hurt performance?) if the key we are setting is the entire schema itself? ... is it not possible to have a simpler key?

Reply:
No, the key is just the internal object id. And it's a weak map which won't prevent the schema from being GCed if it's no longer referenced anywhere else.